mail_base/static/src/js/base.js

```diff
@@ -0,0 +1,1083 @@
+        chat_manager.bus.trigger('update_needaction', needaction_counter);
+    },
+
+    on_mark_as_unread_notification: function (data) {
+        _.each(data.message_ids, function (message_id) {
+            var message = _.findWhere(messages, { id: message_id });
+            if (message) {
+                this.invalidate_caches(message.channel_ids);
+                this.add_channel_to_message(message, 'channel_inbox');
+                this.add_to_cache(message, []);
+            }
+        });
+        var channel_inbox = chat_manager.get_channel('channel_inbox');
+        channel_inbox.cache = _.pick(channel_inbox.cache, "[]");
+
+        _.each(data.channel_ids, function (channel_id) {
+            var channel = chat_manager.get_channel(channel_id);
+            if (channel) {
+                channel.needaction_counter += data.message_ids.length;
+            }
+        });
+        needaction_counter += data.message_ids.length;
+        chat_manager.bus.trigger('update_needaction', needaction_counter);
+    },
+
+    on_channel_seen_notification: function (data) {
+        var channel = chat_manager.get_channel(data.id);
+        if (channel) {
+            channel.last_seen_message_id = data.last_message_id;
+            if (channel.unread_counter) {
+                this.update_channel_unread_counter(channel, 0);
+            }
+        }
+    },
+
+    on_chat_session_notification: function (chat_session) {
+        var channel;
+        if ((chat_session.channel_type === "channel") && (chat_session.state === "open")) {
+            this.add_channel(chat_session, {autoswitch: false});
+            if (!chat_session.is_minimized && chat_session.info !== 'creation') {
+                web_client.do_notify(_t("Invitation"), _t("You have been invited to: ") + chat_session.name);
+            }
+        }
+        // partner specific change (open a detached window for example)
+        if ((chat_session.state === "open") || (chat_session.state === "folded")) {
+            channel = chat_session.is_minimized && chat_manager.get_channel(chat_session.id);
+            if (channel) {
+                channel.is_detached = true;
+                channel.is_folded = (chat_session.state === "folded");
+                chat_manager.bus.trigger("open_chat", channel);
+            }
+        } else if (chat_session.state === "closed") {
+            channel = chat_manager.get_channel(chat_session.id);
+            if (channel) {
+                channel.is_detached = false;
+                chat_manager.bus.trigger("close_chat", channel, {keep_open_if_unread: true});
+            }
+        }
+    },
+
+    on_presence_notification: function (data) {
+        var dm = chat_manager.get_dm_from_partner_id(data.id);
+        if (dm) {
+            dm.status = data.im_status;
+            chat_manager.bus.trigger('update_dm_presence', dm);
+        }
+    },
+
+    start: function(){
```

Comment:
move this function outside of chat_manager and replace *this* to *chat_manager*

     function init(){
        chat_manager.add_channel({
            id: "channel_inbox",
            name: _t("Inbox"),
            type: "static"
         }, { display_needactions: true })
     }

We have to do it to show, that *init* function cannot be inherit this way.

Also, add a comment
    
    // In order to extend *init* use chat_manager.is_ready Derrered object. See example in mail_arhive module